nix/pkgs/libspdk/default.nix

```diff
@@ -50,20 +50,19 @@ let
     ];
 
     configureFlags = [
-      "--target-arch=nehalem"
+      "--target-arch=native"
       "--without-isal"
       "--with-iscsi-initiator"
-      "--with-crypto"
       "--with-uring"
+      # "--with-crypto"
```

Comment:
I would prefer not to do this. :|

Reply:
So it looks like `--with-crypto` depends on `aes` which is not available on aarch64.

```bash
[nix-shell:~/git/spdk/spdk]$ ./configure --with-crypto
Using default SPDK env in /home/ana/git/spdk/spdk/lib/env_dpdk
Using default DPDK in /home/ana/git/spdk/spdk/dpdk/build
Configuring ISA-L (logfile: /home/ana/git/spdk/spdk/isa-l/spdk-isal.log)...done.
Creating mk/config.mk...done.
Creating mk/cc.flags.mk...done.
Type 'make' to build.

[nix-shell:~/git/spdk/spdk]$ make --trace -j12
Makefile:116: update target 'build_dir' due to: mk/cc.mk
mkdir -p build/lib
Makefile:123: update target 'include/spdk/config.h' due to: mk/config.mk scripts/genconfig.py
echo "#ifndef SPDK_CONFIG_H" > include/spdk/config.h.tmp; \
echo "#define SPDK_CONFIG_H" >> include/spdk/config.h.tmp; \
scripts/genconfig.py  -j12 --jobserver-auth=3,4 --trace --no-print-directory >> include/spdk/config.h.tmp; \
echo "#endif /* SPDK_CONFIG_H */" >> include/spdk/config.h.tmp; \
cmp -s include/spdk/config.h.tmp include/spdk/config.h || mv include/spdk/config.h.tmp include/spdk/config.h ; \
rm -f include/spdk/config.h.tmp
mkdir -p build/bin
mkdir -p build/fio
mkdir -p build/examples
mkdir -p build/include/spdk
/home/ana/git/spdk/spdk/mk/spdk.subdirs.mk:44: update target 'ipsecbuild' due to: mk/cc.mk build_dir include/spdk/config.h
make -C ipsecbuild S=ipsecbuild all
/home/ana/git/spdk/spdk/mk/spdk.subdirs.mk:44: update target 'isalbuild' due to: mk/cc.mk build_dir include/spdk/config.h
make -C isalbuild S=isalbuild all
/home/ana/git/spdk/spdk/mk/spdk.subdirs.mk:44: update target 'include' due to: mk/cc.mk build_dir include/spdk/config.h
make -C include S=include all
Makefile:50: update target 'all' due to: /home/ana/git/spdk/spdk/isa-l/isa-l
Makefile:47: target 'all' does not exist
make -C /home/ana/git/spdk/spdk/isa-l all CC="gcc" > /dev/null
make -C /home/ana/git/spdk/spdk/intel-ipsec-mb all SHARED=n EXTRA_CFLAGS=-fPIC CC=gcc > /dev/null
Makefile:50: target 'all' does not exist
cp /home/ana/git/spdk/spdk/include/spdk/*.h /home/ana/git/spdk/spdk/build/include/spdk/
gcc: error: unrecognized command line option ‘-maes’
make[2]: *** [Makefile:603: obj/mb_mgr_avx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
gcc: error: unrecognized command line option ‘-maes’
gcc: error: unrecognized command line option ‘-maes’
make[2]: *** [Makefile:613: obj/mb_mgr_avx2.o] Error 1
make[2]: *** [Makefile:593: obj/mb_mgr_sse.o] Error 1
cc1: error: unknown value ‘broadwell’ for ‘-march’
cc1: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8.5-a native
cc1: error: unknown value ‘nehalem’ for ‘-march’
cc1: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8.5-a native
make[2]: *** [Makefile:623: obj/mb_mgr_avx512.o] Error 1
make[2]: *** [Makefile:640: obj/mb_mgr_sse_no_aesni.o] Error 1
^Cmake[2]: *** Deleting file 'obj/gcm256_sse_no_aesni.o'
make[2]: *** Deleting file 'obj/gcm192_sse_no_aesni.o'
make[2]: *** [Makefile:646: obj/gcm256_sse_no_aesni.o] Interrupt
make[2]: *** [Makefile:646: obj/gcm192_sse_no_aesni.o] Interrupt
make[1]: *** [Makefile:47: all] Interrupt
make: *** [/home/ana/git/spdk/spdk/mk/spdk.subdirs.mk:44: ipsecbuild] Interrupt


[nix-shell:~/git/spdk/spdk]$ uname -a
Linux gizmo 5.10.5 #1-NixOS SMP Tue Jan 1 00:00:00 UTC 1980 aarch64 GNU/Linux

[nix-shell:~/git/spdk/spdk]$ cat /proc/cpuinfo 
processor	: 0
BogoMIPS	: 50.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3
```

Reply:
When I compare the `rte_build_config.h` using spdk:

```
Configuring rte_build_config.h using configuration
Message: 
=================
Libraries Enabled
=================

libs:
	kvargs, telemetry, eal, ring, rcu, mempool, mbuf, net, 
	meter, ethdev, pci, cmdline, hash, timer, compressdev, cryptodev, 
	power, reorder, security, vhost, 

Message: 
===============
Drivers Enabled
===============

common:
	
bus:
	pci, vdev, 
mempool:
	ring, 
net:
	
raw:
	
crypto:
	
compress:
	
regex:
	
vdpa:
	
event:
	
baseband:
	

Message: 
=================
Content Skipped
=================

libs:
	metrics:	not needed by SPDK
	acl:	not needed by SPDK
	bbdev:	not needed by SPDK
	bitratestats:	not needed by SPDK
	cfgfile:	not needed by SPDK
	distributor:	not needed by SPDK
	efd:	not needed by SPDK
	eventdev:	not needed by SPDK
	gro:	not needed by SPDK
	gso:	not needed by SPDK
	ip_frag:	not needed by SPDK
	jobstats:	not needed by SPDK
	kni:	not needed by SPDK
	latencystats:	not needed by SPDK
	lpm:	not needed by SPDK
	member:	not needed by SPDK
	pdump:	not needed by SPDK
	rawdev:	not needed by SPDK
	regexdev:	not needed by SPDK
	rib:	not needed by SPDK
	sched:	not needed by SPDK
	stack:	not needed by SPDK
	ipsec:	not needed by SPDK
	fib:	not needed by SPDK
	port:	not needed by SPDK
	table:	not needed by SPDK
	pipeline:	not needed by SPDK
	flow_classify:	not needed by SPDK
	bpf:	not needed by SPDK
	graph:	not needed by SPDK
	node:	not needed by SPDK
	
drivers:
	common/cpt:	Explicitly disabled via build config
	common/dpaax:	Explicitly disabled via build config
	common/iavf:	Explicitly disabled via build config
	common/mvep:	Explicitly disabled via build config
	common/octeontx:	Explicitly disabled via build config
	common/octeontx2:	Explicitly disabled via build config
	common/qat:	Explicitly disabled via build config
	bus/dpaa:	Explicitly disabled via build config
	bus/fslmc:	Explicitly disabled via build config
	bus/ifpga:	Explicitly disabled via build config
	bus/vmbus:	Explicitly disabled via build config
	common/mlx5:	Explicitly disabled via build config
	mempool/bucket:	Explicitly disabled via build config
	mempool/dpaa:	Explicitly disabled via build config
	mempool/dpaa2:	Explicitly disabled via build config
	mempool/octeontx:	Explicitly disabled via build config
	mempool/octeontx2:	Explicitly disabled via build config
	mempool/stack:	Explicitly disabled via build config
	net/af_packet:	Explicitly disabled via build config
	net/af_xdp:	Explicitly disabled via build config
	net/ark:	Explicitly disabled via build config
	net/atlantic:	Explicitly disabled via build config
	net/avp:	Explicitly disabled via build config
	net/axgbe:	Explicitly disabled via build config
	net/bonding:	Explicitly disabled via build config
	net/bnx2x:	Explicitly disabled via build config
	net/bnxt:	Explicitly disabled via build config
	net/cxgbe:	Explicitly disabled via build config
	net/dpaa:	Explicitly disabled via build config
	net/dpaa2:	Explicitly disabled via build config
	net/e1000:	Explicitly disabled via build config
	net/ena:	Explicitly disabled via build config
	net/enetc:	Explicitly disabled via build config
	net/enic:	Explicitly disabled via build config
	net/failsafe:	Explicitly disabled via build config
	net/fm10k:	Explicitly disabled via build config
	net/i40e:	Explicitly disabled via build config
	net/hinic:	Explicitly disabled via build config
	net/hns3:	Explicitly disabled via build config
	net/iavf:	Explicitly disabled via build config
	net/ice:	Explicitly disabled via build config
	net/igc:	Explicitly disabled via build config
	net/ipn3ke:	Explicitly disabled via build config
	net/ixgbe:	Explicitly disabled via build config
	net/kni:	Explicitly disabled via build config
	net/liquidio:	Explicitly disabled via build config
	net/memif:	Explicitly disabled via build config
	net/mlx4:	Explicitly disabled via build config
	net/mlx5:	Explicitly disabled via build config
	net/mvneta:	Explicitly disabled via build config
	net/mvpp2:	Explicitly disabled via build config
	net/netvsc:	Explicitly disabled via build config
	net/nfb:	Explicitly disabled via build config
	net/nfp:	Explicitly disabled via build config
	net/null:	Explicitly disabled via build config
	net/octeontx:	Explicitly disabled via build config
	net/octeontx2:	Explicitly disabled via build config
	net/pcap:	Explicitly disabled via build config
	net/pfe:	Explicitly disabled via build config
	net/qede:	Explicitly disabled via build config
	net/ring:	Explicitly disabled via build config
	net/sfc:	Explicitly disabled via build config
	net/softnic:	Explicitly disabled via build config
	net/szedata2:	Explicitly disabled via build config
	net/tap:	Explicitly disabled via build config
	net/thunderx:	Explicitly disabled via build config
	net/vdev_netvsc:	Explicitly disabled via build config
	net/vhost:	Explicitly disabled via build config
	net/virtio:	Explicitly disabled via build config
	net/vmxnet3:	Explicitly disabled via build config
	raw/dpaa2_cmdif:	Explicitly disabled via build config
	raw/dpaa2_qdma:	Explicitly disabled via build config
	raw/ifpga:	Explicitly disabled via build config
	raw/ioat:	Explicitly disabled via build config
	raw/ntb:	Explicitly disabled via build config
	raw/octeontx2_dma:	Explicitly disabled via build config
	raw/octeontx2_ep:	Explicitly disabled via build config
	raw/skeleton:	Explicitly disabled via build config
	crypto/aesni_gcm:	Explicitly disabled via build config
	crypto/aesni_mb:	Explicitly disabled via build config
	crypto/armv8:	Explicitly disabled via build config
	crypto/caam_jr:	Explicitly disabled via build config
	crypto/ccp:	Explicitly disabled via build config
	crypto/dpaa_sec:	Explicitly disabled via build config
	crypto/dpaa2_sec:	Explicitly disabled via build config
	crypto/kasumi:	Explicitly disabled via build config
	crypto/mvsam:	Explicitly disabled via build config
	crypto/nitrox:	Explicitly disabled via build config
	crypto/null:	Explicitly disabled via build config
	crypto/octeontx:	Explicitly disabled via build config
	crypto/octeontx2:	Explicitly disabled via build config
	crypto/openssl:	Explicitly disabled via build config
	crypto/qat:	Explicitly disabled via build config
	crypto/scheduler:	Explicitly disabled via build config
	crypto/snow3g:	Explicitly disabled via build config
	crypto/virtio:	Explicitly disabled via build config
	crypto/zuc:	Explicitly disabled via build config
	compress/isal:	Explicitly disabled via build config
	compress/octeontx:	Explicitly disabled via build config
	compress/qat:	Explicitly disabled via build config
	compress/zlib:	Explicitly disabled via build config
	regex/mlx5:	Explicitly disabled via build config
	vdpa/ifc:	Explicitly disabled via build config
	vdpa/mlx5:	Explicitly disabled via build config
	event/dpaa:	Explicitly disabled via build config
	event/dpaa2:	Explicitly disabled via build config
	event/octeontx2:	Explicitly disabled via build config
	event/opdl:	Explicitly disabled via build config
	event/skeleton:	Explicitly disabled via build config
	event/sw:	Explicitly disabled via build config
	event/dsw:	Explicitly disabled via build config
	event/octeontx:	Explicitly disabled via build config
	baseband/null:	Explicitly disabled via build config
	baseband/turbo_sw:	Explicitly disabled via build config
	baseband/fpga_lte_fec:	Explicitly disabled via build config
	baseband/fpga_5gnr_fec:	Explicitly disabled via build config
	

Build targets in project: 123
```

And dpdk:

```bash
Configuring rte_build_config.h using configuration
Message: 
=================
Libraries Enabled
=================

libs:
	kvargs, telemetry, eal, ring, rcu, mempool, mbuf, net, 
	meter, ethdev, pci, cmdline, hash, timer, compressdev, cryptodev, 
	power, reorder, security, vhost, 

Message: 
===============
Drivers Enabled
===============

common:
	cpt, dpaax, iavf, octeontx, octeontx2, 
bus:
	pci, vdev, vmbus, 
mempool:
	bucket, octeontx, octeontx2, ring, 
net:
	af_packet, ark, atlantic, avp, axgbe, bnxt, cxgbe, e1000, 
	ena, enetc, enic, failsafe, fm10k, i40e, hinic, hns3, 
	iavf, ice, igc, ixgbe, liquidio, memif, netvsc, nfp, 
	null, pfe, qede, ring, thunderx, vdev_netvsc, vhost, virtio, 
	vmxnet3, 
raw:
	
crypto:
	nitrox, null_crypto, octeontx_crypto, octeontx2_crypto, crypto_scheduler, virtio_crypto, 
compress:
	isal, octeontx_compress, qat, 
regex:
	
vdpa:
	ifc, 
event:
	
baseband:
	

Message: 
=================
Content Skipped
=================

libs:
	metrics:	not needed by SPDK
	acl:	not needed by SPDK
	bbdev:	not needed by SPDK
	bitratestats:	not needed by SPDK
	cfgfile:	not needed by SPDK
	distributor:	not needed by SPDK
	efd:	not needed by SPDK
	eventdev:	not needed by SPDK
	gro:	not needed by SPDK
	gso:	not needed by SPDK
	ip_frag:	not needed by SPDK
	jobstats:	not needed by SPDK
	kni:	not needed by SPDK
	latencystats:	not needed by SPDK
	lpm:	not needed by SPDK
	member:	not needed by SPDK
	pdump:	not needed by SPDK
	rawdev:	not needed by SPDK
	regexdev:	not needed by SPDK
	rib:	not needed by SPDK
	sched:	not needed by SPDK
	stack:	not needed by SPDK
	ipsec:	not needed by SPDK
	fib:	not needed by SPDK
	port:	not needed by SPDK
	table:	not needed by SPDK
	pipeline:	not needed by SPDK
	flow_classify:	not needed by SPDK
	bpf:	not needed by SPDK
	graph:	not needed by SPDK
	node:	not needed by SPDK
	
drivers:
	common/mvep:	missing dependency, "libmusdk"
	bus/dpaa:	Missing internal dependency, "eventdev"
	bus/fslmc:	Missing internal dependency, "eventdev"
	bus/ifpga:	Missing internal dependency, "rawdev"
	common/mlx5:	missing dependency, "mlx5"
	mempool/dpaa:	Missing internal dependency, "bus_dpaa"
	mempool/dpaa2:	Missing internal dependency, "bus_fslmc"
	mempool/stack:	Missing internal dependency, "stack"
	net/af_xdp:	missing dependency, "libbpf"
	net/bonding:	Missing internal dependency, "ip_frag"
	net/bnx2x:	missing dependency, "zlib"
	net/dpaa:	Missing internal dependency, "mempool_dpaa"
	net/dpaa2:	Missing internal dependency, "mempool_dpaa2"
	net/ipn3ke:	missing dependency, "libfdt"
	net/kni:	missing dependency, DPDK KNI library
	net/mlx4:	missing dependency, "mlx4"
	net/mlx5:	Missing internal dependency, "common_mlx5"
	net/mvneta:	missing dependency, "libmusdk"
	net/mvpp2:	missing dependency, "libmusdk"
	net/nfb:	missing dependency, "libnfb"
	net/octeontx:	Missing internal dependency, "eventdev"
	net/octeontx2:	Missing internal dependency, "eventdev"
	net/pcap:	missing dependency, "libpcap"
	net/sfc:	only supported on x86_64
	net/softnic:	Missing internal dependency, "sched"
	net/szedata2:	missing dependency, "libsze2"
	net/tap:	Missing internal dependency, "gso"
	raw/dpaa2_cmdif:	missing dependency, DPDK DPAA2 mempool driver
	raw/dpaa2_qdma:	missing dependency, DPDK DPAA2 mempool driver
	raw/ifpga:	missing dependency, "libfdt"
	raw/ioat:	only supported on x86
	raw/ntb:	Missing internal dependency, "rawdev"
	raw/octeontx2_dma:	Missing internal dependency, "rawdev"
	raw/octeontx2_ep:	Missing internal dependency, "rawdev"
	raw/skeleton:	Missing internal dependency, "rawdev"
	crypto/aesni_gcm:	missing dependency, "libIPSec_MB"
	crypto/aesni_mb:	missing dependency, "libIPSec_MB"
	crypto/armv8:	missing dependency, "armv8_crypto"
	crypto/caam_jr:	Missing internal dependency, "bus_dpaa"
	crypto/ccp:	missing dependency, "libcrypto"
	crypto/dpaa_sec:	Missing internal dependency, "mempool_dpaa"
	crypto/dpaa2_sec:	Missing internal dependency, "mempool_dpaa2"
	crypto/kasumi:	missing dependency, "libIPSec_MB"
	crypto/mvsam:	missing dependency, "libmusdk"
	crypto/openssl:	missing dependency, "libcrypto"
	crypto/snow3g:	missing dependency, "libIPSec_MB"
	crypto/zuc:	missing dependency, "libIPSec_MB"
	compress/zlib:	missing dependency, "zlib"
	regex/mlx5:	Missing internal dependency, "regexdev"
	vdpa/mlx5:	Missing internal dependency, "sched"
	event/dpaa:	Missing internal dependency, "pmd_dpaa_sec"
	event/dpaa2:	Missing internal dependency, "pmd_dpaa2_sec"
	event/octeontx2:	Missing internal dependency, "pmd_octeontx2"
	event/opdl:	Missing internal dependency, "eventdev"
	event/skeleton:	Missing internal dependency, "eventdev"
	event/sw:	Missing internal dependency, "eventdev"
	event/dsw:	Missing internal dependency, "eventdev"
	event/octeontx:	Missing internal dependency, "pmd_octeontx"
	baseband/null:	Missing internal dependency, "bbdev"
	baseband/turbo_sw:	Missing internal dependency, "bbdev"
	baseband/fpga_lte_fec:	Missing internal dependency, "bbdev"
	baseband/fpga_5gnr_fec:	Missing internal dependency, "bbdev"
```

Reply:
Resolving this as we do not currently use this module, so it's not needed.